frontend/packages/dev-console/src/utils/pipeline-plugin-utils.ts

```diff
@@ -0,0 +1,59 @@
+import * as _ from 'lodash';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { PipelineOverviewItem } from '@console/shared';
+import { PipelineRunModel, PipelineModel } from '../models';
+import { Pipeline, PipelineRun } from './pipeline-augment';
+
+// label to get the pipelines
+export const PIPELINE_REF = 'app.openshift.io/pipeline-ref';
+
+export const tknPipelineAndPipelineRunsResources = (namespace: string): FirehoseResource[] => {
+  const resources = [
+    {
+      isList: true,
+      kind: referenceForModel(PipelineRunModel),
+      namespace,
+      prop: 'pipelineRuns',
+      optional: true,
+    },
+    {
+      isList: true,
+      kind: referenceForModel(PipelineModel),
+      namespace,
+      prop: 'pipelines',
+      optional: true,
+    },
+  ];
+  return resources;
+};
+
+const getPipelineRunsForPipeline = (pipeline: Pipeline, props): PipelineRun[] => {
+  if (props && !props.pipelineRuns) return undefined;
```

Comment:
Couple things here... typically you don't return `undefined` unless you're trying to trigger default values - which isn't what I am seeing here. You're making a call on line 56 and your type specifies it's not optional.

And the other thing is, I am pretty sure there is a logic flaw here:
```suggestion
  if (!props || !props.pipelineRuns) return undefined;
```
Otherwise on line 33 you're going to have an issue if `props` is not an object. If `props` cannot be null (or another falsey value) then you don't need to check against it in this if-condition.

Reply:
Thanks for noticing this. I was also a bit skeptical about the condition fixed in https://github.com/openshift/console/pull/3014/commits/61c377e657d2ed8303b93b45260f10904fda47f2